client_test.go

```diff
@@ -227,7 +228,7 @@ func TestTCPClient(t *testing.T) {
 	require.NoError(t, err)
 
 	client, err := NewClient(&ClientConfig{
-		Conn:           NewSTUNConn(conn),
```

Comment:
Isn't this breaking the API by removing `NewSTUNConn`?

I agree that isnt a nice API. And I would much like to see it moved to `pion/stun` (see https://github.com/pion/turn/issues/308 for a related issue).

I think we currently have a lot STUN related code in `pion/turn`. E.g. `PerformTransaction` which would be better located in `pion/stun`. However, this is a different issue which we should not address in this PR.